x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportInferModelAction.java

```diff
@@ -24,20 +27,28 @@
 
     private final ModelLoadingService modelLoadingService;
     private final Client client;
+    private final XPackLicenseState licenseState;
 
     @Inject
     public TransportInferModelAction(TransportService transportService,
                                      ActionFilters actionFilters,
                                      ModelLoadingService modelLoadingService,
-                                     Client client) {
+                                     Client client,
+                                     XPackLicenseState licenseState) {
         super(InferModelAction.NAME, transportService, actionFilters, InferModelAction.Request::new);
         this.modelLoadingService = modelLoadingService;
         this.client = client;
+        this.licenseState = licenseState;
     }
 
     @Override
     protected void doExecute(Task task, InferModelAction.Request request, ActionListener<InferModelAction.Response> listener) {
 
+        if (licenseState.isMachineLearningAllowed() == false) {
```

Comment:
Do other actions (get, delete) already have this license check?

Reply:
No, I don't think they should either. They don't really provide licensed value IMO. This is similar to how we treat anomaly jobs as well.